src/script/components/manifest-options.ts

```diff
@@ -412,14 +413,37 @@ export class AppManifest extends LitElement {
   constructor() {
     super();
 
-    manifestEmitter.addEventListener(AppEvents.manifestUpdate, async () => {
-
-      this.manifest = await getManifest();
+    manifestEmitter.addEventListener(AppEvents.manifestUpdate, async (maniUpdates: any) => {
+      console.log('maniUpdates', maniUpdates);
+      if (maniUpdates) {
+        this.manifest = maniUpdates.detail;
+      }
     });
   }
 
   async firstUpdated() {
-    this.manifest = await getManifest();
+    try {
```

Comment:
is there a way to abstract this code block out to a function within the service? The component is doing a lot of work that is data-centric, and it looks like it starting to look more like a 'controller' to use MVC terms. The manifest-options doesn't really need to worry about if the manifest is generated or from the site, but good change though! Much better than depending on an update, like I was doing. Sorry I didn't think of it earlier.

Reply:
Yep it can! Wanted to get it working for now and then re-factor to some better code after June (: . Also, no apology needed at all 😊